### PR TITLE
Fix SonarC#: Parameterized rules are not read properly

### LINKED ISSUE #791: Fix SonarC#: Parameterized rules are not read properly
### Description

Parameterized rules are no longer working properly, the default value is always used (regardless of the configuration on the server).

### Expected behavior

`warning S1067: Reduce the number of conditional operators (4) used in the expression (maximum allowed 1).`

### Actual behavior

` warning S1067: Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).`
Which is the default value of the rule.

### Known workarounds

Rollback to previous versions.

### Related information

* SonarC# Version: 6.4

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #791

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/ParameterLoader.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ private static ImmutableList<RuleParameterValues> ParseParameters(XDocument xml)
 
             var propertyParameterPairs = parameteredAnalyzer.GetType()
                 .GetTypeInfo()
-                .DeclaredProperties
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/ParameterLoader.cs]: Was not taking properties of base classes.

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj [hunk 2]
```diff
@@ -129,12 +137,16 @@
     </When>
     <Otherwise>
       <ItemGroup>
-        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework" />
+        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework">
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj]: Automatic change. Not sure whether to keep it or not.

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj [hunk 3]
```diff
@@ -129,12 +137,16 @@
...
-        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework" />
+        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework">
+          <Private>False</Private>
+        </Reference>
       </ItemGroup>
     </Otherwise>
   </Choose>
   <ItemGroup>
-    <None Include="packages.SonarAnalyzer.UnitTest.config" />
+    <None Include="packages.SonarAnalyzer.UnitTest.config">
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj]: Automatic change. Not sure whether to keep it or not.

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTests.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class ParameterLoaderTests
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTests.cs]: I am only testing the explicit part (i.e. no test of an invalid xml format, file not found...)

Comment (michalb-sonar) [reply sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTests.cs]: Why do you mean by "explicit"?

Why don't you test for invalid xml/file not found? That would be useful to ensure we fail gracefully.

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj (hunk 2 above)
Comment (michalb-sonar) [reply sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj]: I'd remove, as it makes the diff shorter. But don't mind either way.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/ParameterLoader.cs (hunk 1 above)
Comment (michalb-sonar) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/ParameterLoader.cs]: Is that the fix? Why do we need all the other changes then? ;)

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTests.cs (hunk 4 above)
Comment (Evangelink) [reply sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTests.cs]: What I mean is that the code is not making any check and so will fail with the default ways so I don't think it is worth testing these cases.

**Pull request closed without merging**